### PR TITLE
Implement temperature on aovx

### PR DESCRIPTION
**Pull request by jcardus**
https://github.com/AOVX-Corporation/AOVX_Share/blob/main/Document/en/01_G%20series/AOVX_G%26A_Series%20Cloud%20Platform%20Protocol.pdf

### CODE REVIEW

File: src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java [hunk 1]
```diff
@@ -699,6 +699,14 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
                         }
                     }
                     break;
+                case 0xF6:
+                    buf.readUnsignedByte(); // data type
+                    int fieldMask = buf.readUnsignedByte();
+                    buf.readUnsignedShort(); // light
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java:705]: Shouldn't light also be behind a flag?

Comment (jcardus) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java:705]: they also provide an excel which makes it easier to understand:

https://github.com/AOVX-Corporation/AOVX_Share/blob/main/Document/en/01_G%20series/AOVX_G%26A_Series%20JTT808%20Protocol.xlsx

Screenshot 2024-08-24 at 22 21 43

Comment (tananaev) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java:705]: I don't think you answered my question.

Comment (jcardus) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java]: you are totally right

File: src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java [hunk 2]
```diff
@@ -699,6 +699,16 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
...
                     }
                     break;
+                case 0xF6:
+                    buf.readUnsignedByte(); // data type
+                    int fieldMask = buf.readUnsignedByte();
+                    if (BitUtil.check(fieldMask, 0)) {
+                        buf.readUnsignedShort(); // light
+                    }
+                    if (BitUtil.check(fieldMask, 1)) {
+                        position.set(Position.PREFIX_TEMP + 1, buf.readUnsignedShort() * 0.1);
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java:709]: Can this temperature be negative? Can we add a test case?

Comment (jcardus) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java]: Probably yes, you mean a test case with negative temperature?

Comment (tananaev) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java]: Yes.

Comment (jcardus) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java]: I'll put the device in the freezer to test.

Comment (jcardus) [reply src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java]: it works:

Captura de ecrã 2024-08-25 à(s) 00 56 38

File: src/test/java/org/traccar/protocol/HuabaoProtocolDecoderTest.java [hunk 3]
```diff
@@ -11,6 +11,14 @@ public void testDecode() throws Exception {
 
         var decoder = inject(new HuabaoProtocolDecoder(null));
 
+        verifyAttribute(decoder, binary(
```
Comment (tananaev) [inline src/test/java/org/traccar/protocol/HuabaoProtocolDecoderTest.java:14]: One test is probably enough. Let's just keep negative temperature. It's a better test IMO.

**Comment by tananaev** [comment]
Merged, thank you.

**Pull request merged**